### PR TITLE
WIP ClassificationScoreVisualizers should return accuracy

### LINKED ISSUE #361: ClassificationScoreVisualizers should return accuracy
See #358 and #213  -- classification score visualizers should return accuracy when `score()` is called. If F1 or accuracy is not in the figure it should also be included in the figure.

### PR DESCRIPTION
**Pull request by abatula**
Fixes #361 

Checking my understanding of the issue: ClassificationScoreVisualizer currently does not have an implemented `score()` method. It needs to be created, should return the score function, and should add the accuracy to the plotted figure if accuracy or F1 are not already plotted.

Changes proposed in this pull request:

-
-
-

@DistrictDataLabs/team-oz-maintainers

### CODE REVIEW

**Comment by lwgray** [comment]
Thank you for joining us for the Yellowbrick Sprint at PyCon Cleveland. And also, thank you for submitting your first PullRequest - oh yeah! 

We will review this asap. We have been quite busy with all the great contributions from sprinters like yourself.  In the meantime, feel free to work on another issue.  We hope you continue to contribute.

**Comment by lwgray** [comment]
@abatula  Is there anything we can do to help you complete this PR?

**Comment by abatula** [comment]
@lwgray Sorry, I almost forgot about this for a bit. I was looking for clarification on what exactly the issue is asking for. Is the following correct?

ClassificationScoreVisualizer currently does not have an implemented score() method. It needs to be created, should return the score function, and should add the accuracy to the plotted figure if accuracy or F1 are not already plotted.

**Comment by rebeccabilbro** [comment]
Hey there @abatula @lwgray ! So excited you are taking a look at this one & also jumping in since @bbengfort is at a conference this week and might not be able to check email regularly.

My understanding is that the high-level requirement of #361 is that we want the classification score visualizers to communicate more raw information about the metric scores. 

This can be achieved in a few different steps, for instance, first by exposing internal estimator attributes like an `F1` score for each classification score visualizers, so that after the user has called `score()`, they will be able to retrieve the metric scores via `self.score_`:


    oz = ConfusionMatrix(LogisticRegression())
    oz.fit(X_train, y_train)
    oz.score(X_test, y_test)
    print(oz.score_)


_Note that this is currently implemented for some (or maybe all?) of the regression score visualizers (see [`PredictionError`](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/regressor/residuals.py#L156) for example)._

Once each classification score visualizer has a `score()` method that returns `self.score_`, we can also upgrade their plotting routines to superimpose the actual metrics onto the figures (or by adding them to the title, etc) so that the metadata will be visible to the user on `poof()`.

This would be a super high impact upgrade, since it will mean the classification score visualizers will provide users *way* more diagnostic information than we currently do. Let me know if this helps or if I can offer any additional clarification or assistance!

**Comment by rebeccabilbro** [comment]
Hi again @abatula - the new `score` method is looking good! I propose we address the `draw` part of this in a separate PR (since different classification score visualizers may need to plot the score in slightly different ways). So, to finish this PR up, it looks like the next steps are:
 - [ ] update [`ROCAUC`](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/classifier/rocauc.py#L201) `score` method to return `self.score_` instead of `self.estimator.score`
 - [ ] update [`ClassBalance`](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/classifier/class_balance.py#L92) `score` method to call `self.draw()` and return `self.score_`
 - [ ] update  [`ClassPredictionError`](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/classifier/class_balance.py#L258) `score` method to call `self.draw()` and return `self.score_`
 - [ ] update  [`ClassificationReport`](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/classifier/classification_report.py#L141) `score` method to call `self.draw()` and return `self.score_`
 - [ ] update [`ConfusionMatrix`](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/classifier/confusion_matrix.py#L192) `score` method to call `self.draw()` and return `self.score_`
- [ ] check the [classifier tests](https://github.com/DistrictDataLabs/yellowbrick/tree/develop/tests/test_classifier) to see if any of them are testing the `score` methods (if so, they'll need to be updated, if not, we might want to add a test to make sure each of the above returns `self.score_` as expected after its `score` method has been called)

Let me know if I can help with any of the above and thanks again for working on this!

File: yellowbrick/classifier/base.py [hunk 1]
```diff
@@ -110,14 +110,29 @@ def fit(self, X, y=None, **kwargs):
...
+        Returns
+        -------
+        score : float
         """
-        raise NotImplementedError(
-            "ClassificationScoreVisualizer score function not implemented"
-        )
+        self.score_ =  self.estimator.score(X, y, **kwargs)
+
+        y_pred = self.predict(X)
```
Comment (rebeccabilbro) [inline yellowbrick/classifier/base.py]: @abatula - actually, on second thought, can you just go ahead and delete this (unused) line? After that, we can at least go ahead and merge this in and deal with the other moving parts in a new PR later!

Comment (rebeccabilbro) [reply yellowbrick/classifier/base.py]: (line #133 that is)

**Comment by abatula** [comment]
I didn't see any tests for checking score or the classification score visualizer. Should there be a new file, like the one for ClassificationReportTests? Or should they be part of other existing files?

**Comment by rebeccabilbro** [comment]
@bbengfort - looks lilke we could go ahead and merge this in, since it satisfies the immediate need to ensure the `ClassificationScoreVisualizer` conforms to the Scikit-Learn API. Then we can open up a follow-up issue for updating the `ROCAUC`, `ClassificationReport`, etc with  `self.score_` & updating the tests (which it sounds like @abatula is interested in working on). What do you think?

**Comment by bbengfort** [comment]
@abatula @rebeccabilbro I'd like to go ahead with this one - even though it's just a couple of lines, this is a pretty significant change that will help us in the long run. 

@abatula to answer your question, we could create a file called `tests/test_classifier/test_base.py` and put tests for the `ClassificationScoreVisualizer` there. This would be a good thing to do, but not strictly necessary, since we'll want to test that score is returned in the subclasses, e.g. in [test_classification_report.py](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/tests/test_classifier/test_classification_report.py#L47) we would add `test_score_returns_accuracy` to ensure this is happening.

**Comment by rebeccabilbro** [comment]
Going to go ahead and merge this in @abatula!

**Pull request merged**